Modules/Utils/Utils/UserDefault/UserDefaultWrapper.swift

```diff
@@ -0,0 +1,49 @@
+import Foundation
+
+@propertyWrapper
+public struct UserDefault<T: Equatable>: Equatable {
+    private let key: UserDefaultKey
+    private let defaultValue: T
+    private let userDefaults: UserDefaults
+
+    public init(_ key: UserDefaultKey, defaultValue: T, userDefaults: UserDefaults = .standard) {
+        self.key = key
+        self.defaultValue = defaultValue
+        self.userDefaults = userDefaults
+    }
+
+    public var wrappedValue: T {
+        get {
+            guard let value = userDefaults.object(forKey: key.rawValue) else {
+                return defaultValue
+            }
+
+            return value as? T ?? defaultValue
+        }
+        set {
+            if let value = newValue as? OptionalProtocol, value.isNil() {
```

Comment:
Do you really need the `OptionalProtocol`? Why not something like this?
```suggestion
            if let value = newValue as? Optional<T>, value == nil {
```

Reply:
this throughs a warning